lib/phoenix/transports/long_poll.ex

```diff
@@ -72,7 +72,22 @@ defmodule Phoenix.Transports.LongPoll do
   defp publish(conn, server_ref, endpoint, opts) do
     case read_body(conn, []) do
       {:ok, body, conn} ->
-        status = transport_dispatch(endpoint, server_ref, body, opts)
+        # we need to match on both v1 and v2 protocol, as well as wrap for backwards compat
+        batch =
+          if String.starts_with?(body, ~w(["[)) do
+            Phoenix.json_library().decode!(body)
+          else
+            [body]
+          end
```

Comment:
Sorry, I am a bit confused. Why don't we need to `decode!` on the else branch?

Reply:
Ah, I think I got it. We pass the body to the socket.handle_in, which then calls deserialize.

So unfortunately there is something wrong with this pull request in that we are now calling `socket.handle_in({text, op: :text}, ...)` but `text` is already an encoded document. Which means we need to find another way to encode the json document. Maybe we could use [NDJSON](http://ndjson.org/)? And then here it is just a matter of splitting on newlines. We need to check if JSON.stringify guarantees to encode newlines as `\n`.